### PR TITLE
feat(ONB-102): Crawl Users From Webpage

### PR DESCRIPTION
**Pull request by eececan**
## Description

> Crawl Users From Webpage

### Development and testing

- [x ] Related unit test have been implemented/updated
- [ x] Application changes have been tested locally
- [x ] Related documents have been updated

### Security

- [ x] Security impact of the change has been considered
- [x ] Code follows MobileAction's security practices and policies

### Code review 

- [ x] Pull request title follows [conventional commit format](https://mobileaction.atlassian.net/wiki/spaces/ENG/pages/743112854/Conventional+Commit+Messages) and has a JIRA task id

### CODE REVIEW

File: web/src/main/java/co/mobileaction/example/web/repository/IPostRepository.java [hunk 1]
```diff
@@ -10,7 +13,13 @@
  * @date 17.05.2021
  * @time 17:45
  */
+@Transactional
+@Repository
 public interface IPostRepository extends JpaRepository<Post, Long>
 {
     List<Post> findAllByUserId(Long userId);
+
+    @Query("select distinct p.userId from Post p")
```
Comment (eculcu) [inline web/src/main/java/co/mobileaction/example/web/repository/IPostRepository.java:22]: Do you need a native query for this operation? How can we handle with JPA?

File: web/src/main/java/co/mobileaction/example/web/queue/UserResultQueueHandler.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+@RequiredArgsConstructor
+public class UserResultQueueHandler
+{
+    private final AmqpTemplate userResultProblemQueueTemplate;
+    private final IUserResultHandlerService resultHandlerService;
+
+    @RabbitListener(queues = "${messaging.queue.userresult}", containerFactory = "resultQueueListener")
+    public void handleMessage(UserDto result)
+    {
+        log.error("ee");
```
Comment (barisonen) [inline web/src/main/java/co/mobileaction/example/web/queue/UserResultQueueHandler.java]: ee

File: web/src/main/java/co/mobileaction/example/web/repository/IUserRepository.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+ * @author elif ece can
+ * @date 04.07.2024
+ * @time 17:45
+ */
+
+@Transactional
+@Repository
+public interface IUserRepository extends JpaRepository<User, Long>
+{
+    List<User> findAll();
```
Comment (barisonen) [inline web/src/main/java/co/mobileaction/example/web/repository/IUserRepository.java]: you don't need this, the exact method already exists in JpaRepository

File: web/src/test/java/co/mobileaction/example/web/controller/UserControllerTests.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+/**
+ * @author sa
+ * @date 17.05.2021
+ * @time 18:58
+ */
+@WebMvcTest(controllers = UserController.class)
+@ContextConfiguration(classes = UserController.class)
+@WithMockUser(roles = {SecurityUtils.USER})
+public class UserControllerTests extends ControllerTestsBase
```
Comment (barisonen) [inline web/src/test/java/co/mobileaction/example/web/controller/UserControllerTests.java:29]: These tests fail, if you remove the actual APIs you should remove also the tests.

File: web/src/main/java/co/mobileaction/example/web/repository/IPostRepository.java (hunk 1 above)
Comment (eececan) [reply web/src/main/java/co/mobileaction/example/web/repository/IPostRepository.java:22]: Maybe we can create a post specification for distinct user ids and use the findAll method in our repository.

Comment (eececan) [reply web/src/main/java/co/mobileaction/example/web/repository/IPostRepository.java:22]: Also, JPA naming conventions can be used to write a method findDistinctByUserId .

File: web/src/main/java/co/mobileaction/example/web/queue/UserResultQueueHandler.java (hunk 2 above)
Comment (eececan) [reply web/src/main/java/co/mobileaction/example/web/queue/UserResultQueueHandler.java]: i added the statement for debugging purposes, i will remove it.

File: web/src/test/java/co/mobileaction/example/web/controller/UserControllerTests.java (hunk 4 above)
Comment (eececan) [reply web/src/test/java/co/mobileaction/example/web/controller/UserControllerTests.java:29]: i will remove them, thank you.

File: web/src/main/java/co/mobileaction/example/web/repository/IPostRepository.java (hunk 1 above)
Comment (barisonen) [reply web/src/main/java/co/mobileaction/example/web/repository/IPostRepository.java:22]: @eececan trying these and other different approaches while also checking the queries performed would be beneficial

**Pull request closed without merging**